The gRPC log is output to the wrong file

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

1. Start TiDB with the following config:
```
[log.file]
max-size = 1
```

2. Enable TopSQL or set `GRPC_DEBUG` environment variable to let gRPC keep printing 
```
[ERROR] [grpclogger.go:116] ["[transport]transport: Got too many pings from the client, closing the connection."] [system=grpc] [grpc_log=true]"
[ERROR] [grpclogger.go:116] ["[transport]transport: loopyWriter.run returning. Err: transport: Connection closing"] [system=grpc] [grpc_log=true]
```

3. Add workload to cluster to populate the log file.
4. When the log file is rotated, gRPC error message should be output to the origin log file.

### 2. What did you expect to see? (Required)

New error messages from gRPC should be output to "tidb.log".

### 3. What did you see instead (Required)

New error messages from gRPC are output to the rotated log file.

From tidb-2022-11-07T16-14-51.727.log:
```
[2022/11/07 16:15:35.553 +08:00] [ERROR] [grpclogger.go:116] ["[transport]transport: Got too many pings from the client, closing the connection."] [system=grpc] [grpc_log=true]
[2022/11/07 16:15:35.554 +08:00] [ERROR] [grpclogger.go:116] ["[transport]transport: loopyWriter.run returning. Err: transport: Connection closing"] [system=grpc] [grpc_log=true]
[2022/11/07 16:16:45.557 +08:00] [ERROR] [grpclogger.go:116] ["[transport]transport: Got too many pings from the client, closing the connection."] [system=grpc] [grpc_log=true]
[2022/11/07 16:16:45.558 +08:00] [ERROR] [grpclogger.go:116] ["[transport]transport: loopyWriter.run returning. Err: transport: Connection closing"] [system=grpc] [grpc_log=true]
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
30a76cef65

--- a/DEPS.bzl
+++ b/DEPS.bzl
@@ -2898,8 +2898,8 @@ def go_deps():
         name = "com_github_pingcap_log",
         build_file_proto_mode = "disable_global",
         importpath = "github.com/pingcap/log",
-        sum = "h1:T7e5Low0BU2ZazI2dz2mh3W1qv+w8wtvq1YR8DneA0c=",
-        version = "v1.1.1-0.20221110065318-21a4942860b3",
+        sum = "h1:crhkw6DD+07Bg1wYhW5Piw+kYNKZqFQqfC2puUf6gMI=",
+        version = "v1.1.1-0.20221116035753-734d527bc87c",
     )
     go_repository(
         name = "com_github_pingcap_sysutil",

--- a/go.mod
+++ b/go.mod
@@ -69,7 +69,7 @@ require (
 	github.com/pingcap/failpoint v0.0.0-20220423142525-ae43b7f4e5c3
 	github.com/pingcap/fn v0.0.0-20200306044125-d5540d389059
 	github.com/pingcap/kvproto v0.0.0-20221103025916-e7e21f0e9cd9
-	github.com/pingcap/log v1.1.1-0.20221110065318-21a4942860b3
+	github.com/pingcap/log v1.1.1-0.20221116035753-734d527bc87c
 	github.com/pingcap/sysutil v0.0.0-20220114020952-ea68d2dbf5b4
 	github.com/pingcap/tidb/parser v0.0.0-20211011031125-9b13dc409c5e
 	github.com/pingcap/tipb v0.0.0-20221020071514-cd933387bcb5

--- a/util/logutil/log.go
+++ b/util/logutil/log.go
@@ -111,37 +111,30 @@ func InitLogger(cfg *LogConfig) error {
 		return errors.Trace(err)
 	}
 
-	_, _, err = initGRPCLogger(cfg)
-	if err != nil {
-		return errors.Trace(err)
-	}
-
+	initGRPCLogger(gl)
 	return nil
 }
 
-func initGRPCLogger(cfg *LogConfig) (*zap.Logger, *log.ZapProperties, error) {
-	// Copy Config struct by assignment.
-	config := cfg.Config
-	var l *zap.Logger
-	var err error
-	var prop *log.ZapProperties
+func initGRPCLogger(gl *zap.Logger) {
+	level := zapcore.ErrorLevel
+	verbosity := 0
 	if len(os.Getenv("GRPC_DEBUG")) > 0 {
-		config.Level = "debug"
-		l, prop, err = log.InitLogger(&config, zap.AddStacktrace(zapcore.FatalLevel))
-		if err != nil {
-			return nil, nil, errors.Trace(err)
-		}
-		gzap.ReplaceGrpcLoggerV2WithVerbosity(l, 999)
-	} else {
-		config.Level = "error"
-		l, prop, err = log.InitLogger(&config, zap.AddStacktrace(zapcore.FatalLevel))
-		if err != nil {
-			return nil, nil, errors.Trace(err)
-		}
-		gzap.ReplaceGrpcLoggerV2(l)
+		verbosity = 999
+		level = zapcore.DebugLevel
 	}
 
-	return l, prop, nil
+	newgl := gl.WithOptions(zap.WrapCore(func(core zapcore.Core) zapcore.Core {
+		oldcore, ok := core.(*log.TextIOCore)
+		if !ok {
+			return oldcore
+		}
+		newcore := oldcore.Clone()
+		leveler := zap.NewAtomicLevel()
+		leveler.SetLevel(level)
+		newcore.LevelEnabler = leveler
+		return newcore
+	}))
+	gzap.ReplaceGrpcLoggerV2WithVerbosity(newgl, verbosity)
 }
 
 // ReplaceLogger replace global logger instance with given log config.

--- a/util/logutil/log_test.go
+++ b/util/logutil/log_test.go
@@ -100,21 +100,6 @@ func TestSetLevel(t *testing.T) {
 	require.Equal(t, zap.DebugLevel, log.GetLevel())
 }
 
-func TestGrpcLoggerCreation(t *testing.T) {
-	level := "info"
-	conf := NewLogConfig(level, DefaultLogFormat, "", EmptyFileLogConfig, false)
-	_, p, err := initGRPCLogger(conf)
-	// assert after init grpc logger, the original conf is not changed
-	require.Equal(t, conf.Level, level)
-	require.NoError(t, err)
-	require.Equal(t, p.Level.Level(), zap.ErrorLevel)
-	os.Setenv("GRPC_DEBUG", "1")
-	defer os.Unsetenv("GRPC_DEBUG")
-	_, newP, err := initGRPCLogger(conf)
-	require.NoError(t, err)
-	require.Equal(t, newP.Level.Level(), zap.DebugLevel)
-}
-
 func TestSlowQueryLoggerCreation(t *testing.T) {
 	level := "Error"
 	conf := NewLogConfig(level, DefaultLogFormat, "", EmptyFileLogConfig, false)

